### PR TITLE
vcd provider: adds support for setting OVF guest properties in vcd_vapp resource

### PR DESCRIPTION
**Pull request by butterflysky**
In VMware, virtual machines can have properties set at create-time which are queryable from within the VM. These properties can be queried by tools within a VM to automate configuration, e.g. automatically applying a specific role manifest within Puppet.

This pull request includes updated documentation and acceptance tests.

### CODE REVIEW

File: builtin/providers/vcd/resource_vcd_vapp.go [hunk 1]
```diff
@@ -207,6 +211,23 @@ func resourceVcdVAppUpdate(d *schema.ResourceData, meta interface{}) error {
 		return fmt.Errorf("Error getting VApp status: %#v", err)
 	}
 
+	if d.HasChange("guest_properties") {
+		_, nraw := d.GetChange("guest_properties")
+		tmp := nraw.(map[string]interface{})
```
Comment (anilybba) [inline builtin/providers/vcd/resource_vcd_vapp.go]: Oops, I don't need a temporary variable here; I can update 218 to check range nraw.(map[string]interface{}) instead.

**Comment by anilybba** [comment]
Hi there @radeksimko, thanks for tagging this pull request for me! Is there something else I should be doing to get this pull request considered / merged for eventual release? I'm trying to follow this one through and am willing to take feedback on anything I should be doing differently. Thank you!

**Comment by Freyert** [comment]
@anilybba I was interested in this because of the mention of OVF. I'm not a VMWare wizard so I'm curious is a v_app the same as a guest_vm?

Any case I think you'll need to move these changes over to https://github.com/terraform-providers/terraform-provider-vsphere

I'm muy interested in getting OVF support into this provider though. Any guidance you can provide or beta from your explorations would be great.

@radeksimko I think this one needs to be closed.

**Comment by butterflysky** [comment]
Hi y'all; I originally intended for this to ship, but I wasn't sure who to ask to review it / how to get it landed and ended up changing jobs and moving house, etc. So this is long forgotten stuff for me that I'm not sure is even relevant 2-3 years later; I don't have time to take it to the finish line anyway. Closing this for now. Thanks!

**Pull request closed without merging**